lessons/basics/modules.md

```diff
@@ -249,8 +288,66 @@ defmacro __using__(_) do
 end
 ```
 
-The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`. It also sets up a module property which we will assume controls Ecto’s behavior.
+The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`.
+It also sets up a module property which we will assume controls Ecto’s behavior.
+
+We can also override a given function that is given to use with `__using__/1`.
+This is done by using `defoverridable`.
+Let's implement this:
+
+```elixir
+iex> defmodule Foo do
+...>   defmacro __using__(_) do
+...>     quote do
+...>       def bar do
+...>         2
+...>       end
+...>       defoverridable [bar: 0]
+...>     end
+...>   end
+...> end
+{:module, Foo,
+ <<. . .>>, {:__using__, 1}}
+iex> defmodule Baz do
+...>   use Foo
+...>   def bar do
+...>     super() + 2
+...>   end
+...> end
+{:module, Baz,
+ <<. . .>>, {:bar, 0}}
+iex> Baz.bar
+4
+```
+
+As you can see, we are able to override the default method `bar` from `use`, and even can call `super` to get the original functionality.
+It is worth noting that `super` must be called with the same number of arguments as the parent method being overridden.
+It also must be placed _after_ the functions being overridden.
+
+### Delegation
+It is possible to define a function that delegates to another module as well.
+If you use `defdelegate` instead of `def`, a given function will be public and invoked from outside the module it is defined in.
```

Comment:
The "invoked from outside the module" portion of this is a little confusing to me.  How about something like:

"Using `defdelegate` we can define a public function and delegate it to another module's function, let's see it in action:"